akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala

```diff
@@ -153,6 +169,7 @@ private[http2] trait Http2StreamHandling { self: GraphStageLogic with LogHelper
     newState match {
       case Closed =>
         streamStates -= streamId
+        if (streamStates.isEmpty && completing) complete()
```

Comment:
I think I would prefer it to keep the whole inlet/outlet logic to the Demux. So, could this be something like 

```
if (streamStates.isEmpty) onAllStreamsClosed()
```

and implement that in the Demux (together with the logic from `tryComplete`). The flag `completed` probably wouldn't be necessary in that case as you could use `isClosed()` there directly.

Reply:
Ah, good idea. Looks cleaner too.